app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,29 @@
+class SessionsController < ApplicationController
+  def front
+    redirect_to videos_path if sign_in?
+  end
+
+  def new
+    redirect_to videos_path if sign_in?
+  end
+
+  def create
+    user = User.find_by_email(params[:email])
+
+    if user && user.authenticate(params[:password])
+      session[:user_id] = user.id
+      redirect_to videos_path
+    else
+      flash[:danger] = "Incorrect email or password, Please try again"
```

Comment:
`flash.now` is prefered vs just `flash` when you render. 

HOMEWORK: Find out why it is better use `.now` method on `flash`